tests/test_simulations.py

```diff
@@ -192,3 +194,237 @@ def test_box_with_zero_gravity(
+            number_of_active_collidable_points_steady_state=4,
+            static_friction_coefficient=1.0,
+            damping_ratio=1.0,
+            max_penetration=0.001,
+        ),
+    )
+
+    # ===========================================
+    # Run the simulation and test the final state
+    # ===========================================
+
+    data_tf = run_simulation(model=model, data_t0=data_t0, dt=0.001, tf=1.0)
+
+    assert data_tf.base_position()[0:2] == pytest.approx(data_t0.base_position()[0:2])
+    assert data_tf.base_position()[2] + max_penetration == pytest.approx(box_height / 2)
+
+
+def test_simulation_with_rigid_contacts(
+    jaxsim_model_box: js.model.JaxSimModel,
+):
+
+    model = jaxsim_model_box
+
+    with model.editable(validate=False) as model:
+
+        model.contact_model = jaxsim.rbda.contacts.RigidContacts.build(
+            terrain=model.terrain,
+        )
+
+    # Initialize the maximum penetration of each collidable point at steady state.
+    # This model is rigid, so we expect (almost) no penetration.
+    max_penetration = 0.000
+
+    # Check jaxsim_model_box@conftest.py.
+    box_height = 0.1
+
+    # Build the data of the model.
+    data_t0 = js.data.JaxSimModelData.build(
+        model=model,
+        base_position=jnp.array([0.0, 0.0, box_height * 2]),
+        velocity_representation=VelRepr.Inertial,
+        # In order to achieve almost no penetration, we need to use a fairly large
+        # Baumgarte stabilization term.
+        contacts_params=js.contact.estimate_good_contacts_parameters(
+            model=model,
+            K=100_000,
+        ),
+    )
+
+    # ===========================================
+    # Run the simulation and test the final state
+    # ===========================================
+
+    data_tf = run_simulation(model=model, data_t0=data_t0, dt=0.001, tf=1.0)
+
+    assert data_tf.base_position()[0:2] == pytest.approx(data_t0.base_position()[0:2])
+    assert data_tf.base_position()[2] + max_penetration == pytest.approx(box_height / 2)
+
+
+def test_simulation_with_relaxed_rigid_contacts(
+    jaxsim_model_box: js.model.JaxSimModel,
+):
+
+    model = jaxsim_model_box
+
+    with model.editable(validate=False) as model:
+
+        model.contact_model = jaxsim.rbda.contacts.RelaxedRigidContacts.build(
+            terrain=model.terrain,
+        )
+
+    # Initialize the maximum penetration of each collidable point at steady state.
+    # This model is quasi-rigid, so we expect (almost) no penetration.
+    max_penetration = 0.000
+
+    # Check jaxsim_model_box@conftest.py.
+    box_height = 0.1
+
+    # Build the data of the model.
+    data_t0 = js.data.JaxSimModelData.build(
+        model=model,
+        base_position=jnp.array([0.0, 0.0, box_height * 2]),
+        velocity_representation=VelRepr.Inertial,
+        # In order to achieve almost no penetration, we need to use a fairly large
+        # Baumgarte stabilization term.
```

Comment:
Maybe it was related to a different contact model
```suggestion
```

Reply:
Thanks! Removed in https://github.com/ami-iit/jaxsim/pull/260/commits/85a98dabe7afa88ef9b7c913d4b8875e2346e244.